### PR TITLE
Fix some IntelliJ findings

### PR DESCRIPTION
**Pull request by koppor**
- Fix bug in TagBar (added itself instead of newTags)
- Wrong number of LOGGER paramters
- Inner class may be static
- Fix comment typo
- Bulk operation can be used instead of iteration
- Arrays.asList with only one element
- Optimized count by using numbers earlier
- Chained append for StringBuilder
- Initialize ArrayList by passing the initial contents in the constructor

### CODE REVIEW

File: src/main/java/org/jabref/gui/fieldeditors/contextmenu/CaseChangeMenu.java [hunk 1]
```diff
@@ -23,8 +23,7 @@ public CaseChangeMenu(final StringProperty text) {
         Objects.requireNonNull(text);
 
         // create menu items, one for each case changer
-        List<Formatter> caseChangers = new ArrayList<>();
-        caseChangers.addAll(Formatters.getCaseChangers());
+        List<Formatter> caseChangers = new ArrayList<>(Formatters.getCaseChangers());
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/fieldeditors/contextmenu/CaseChangeMenu.java]: I found the previous code easier to understand...

File: src/main/java/org/jabref/logic/cleanup/Cleanups.java [hunk 2]
```diff
@@ -36,15 +36,13 @@
         defaultFormatters.add(new FieldFormatterCleanup(InternalField.INTERNAL_ALL_TEXT_FIELDS_FIELD, new ReplaceUnicodeLigaturesFormatter()));
         DEFAULT_SAVE_ACTIONS = new FieldFormatterCleanups(false, defaultFormatters);
 
-        List<FieldFormatterCleanup> recommendedBibTeXFormatters = new ArrayList<>();
-        recommendedBibTeXFormatters.addAll(defaultFormatters);
+        List<FieldFormatterCleanup> recommendedBibTeXFormatters = new ArrayList<>(defaultFormatters);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/cleanup/Cleanups.java]: Here as well.

File: src/main/java/org/jabref/gui/fieldeditors/contextmenu/CaseChangeMenu.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/gui/fieldeditors/contextmenu/CaseChangeMenu.java]: IntelliJ reported that as performance issue. I will revert it nevertheless, as I think it's not that crucial...

**Comment by koppor** [comment]
Reverted the questioned improvements.

File: src/main/java/org/jabref/gui/fieldeditors/contextmenu/CaseChangeMenu.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/fieldeditors/contextmenu/CaseChangeMenu.java]: I actually find the intelij variant easy enough to understand. It's obvious that the constructor of ArrayList can take a collection

**Pull request merged**